cli/internal/cmd/iam_test.go

```diff
@@ -0,0 +1,437 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/edgelesssys/constellation/v2/cli/internal/cloudcmd"
+	"github.com/edgelesssys/constellation/v2/cli/internal/iamid"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/cloudprovider"
+	"github.com/edgelesssys/constellation/v2/internal/config"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+func TestParseIDFile(t *testing.T) {
+	validIAMIDFile := iamid.File{
+		CloudProvider: cloudprovider.GCP,
+		GCPOutput: iamid.GCPFile{
+			ServiceAccountKey: "eyJwcml2YXRlX2tleV9pZCI6Im5vdF9hX3NlY3JldCJ9Cg==", // {"private_key_id":"not_a_secret"}
```

Comment:
IMO this makes sense here since we receive a b64-encoded string from GCP and I want to assert that the part of the code works with a given b64-encoded string

Reply:
We'll still test the same, but I don't want to need to manually re encode the string if some payload format changes.